runelite-client/src/main/java/net/runelite/client/plugins/hydra/HydraPlugin.java

```diff
@@ -0,0 +1,424 @@
+		{
+			// If this is the first observed animation then set it
+			hydra.setHydraAnimation(hydraAnimation);
+		}
+		else
+		{
+			if (!Objects.equals(hydra.getHydraAnimation(), hydraAnimation))
+			{
+				// If the animation switched from range/magic then set it and reset attack count
+				hydra.setHydraAnimation(hydraAnimation);
+				hydra.resetAttackCount();
+			}
+		}
+
+		hydra.updateAttackCount();
+
+		if (!poisonProjectiles.isEmpty())
+		{
+			updatePoisonProjectiles();
+		}
+	}
+
+	/**
+	 * See net.runelite.client.plugins.alchemicalhydra.AlchemicalHydraPlugin
+	 * Copyright (c) 2019, Lucas <https://github.com/lucwousin>
+	 *
+	 * @param event
+	 */
+	@Subscribe
+	private void onProjectileMoved(ProjectileMoved event)
+	{
+		if (Objects.isNull(interactingNpc) || client.getGameCycle() >= event.getProjectile().getStartMovementCycle())
+		{
+			return;
+		}
+
+		Projectile projectile = event.getProjectile();
+
+		int projectileId = projectile.getId();
+
+		if (projectileId == ProjectileID.HYDRA_POISON)
+		{
+			poisonProjectiles.put(event.getPosition(), projectile);
+		}
+	}
+
+	/**
+	 * See net.runelite.client.plugins.alchemicalhydra.AlchemicalHydraPlugin
+	 * Copyright (c) 2019, Lucas <https://github.com/lucwousin>
+	 */
+	private void updatePoisonProjectiles()
+	{
+		Set<LocalPoint> expiredPoisonProjectiles = new HashSet<>();
+
+		for (Map.Entry<LocalPoint, Projectile> entry : poisonProjectiles.entrySet())
+		{
+			if (entry.getValue().getEndCycle() < client.getGameCycle())
+			{
+				expiredPoisonProjectiles.add(entry.getKey());
+			}
+		}
+
+		for (LocalPoint projectileLocalPoint : expiredPoisonProjectiles)
+		{
+			poisonProjectiles.remove(projectileLocalPoint);
+		}
+	}
+
+	boolean isPlayerAtHydraRegion()
+	{
+		Player player = client.getLocalPlayer();
+
+		if (player == null)
+		{
+			return false;
+		}
+
+		WorldPoint worldPoint = player.getWorldLocation();
+
+		if (worldPoint == null)
+		{
+			return false;
+		}
+
+		final int region1 = 5279;
+		final int region2 = 5280;
+
+		int regionId = worldPoint.getRegionID();
+
+		return regionId == region1 || regionId == region2;
+	}
+
+	private static boolean isActorHydra(Actor actor)
+	{
+		final String npcNameHydra = "Hydra";
+
+		return Objects.equals(actor.getName(), npcNameHydra);
+	}
+
+	private void updateInteractingHydra(NPC npc)
+	{
+		if (!Objects.equals(interactingNpc, npc) &&
+			Objects.equals(npc.getInteracting(), client.getLocalPlayer()))
+		{
+			interactingNpc = npc;
+		}
+	}
+
+	private void addHydra(NPC npc)
+	{
+		int npcIndex = npc.getIndex();
+
+		if (!hydras.containsKey(npcIndex))
+		{
+			hydras.put(npcIndex, new Hydra(npc));
+		}
+	}
+
+	private void removeHydra(NPC npc)
+	{
+		int npcIndex = npc.getIndex();
+
+		hydras.remove(npcIndex);
+
+		if (Objects.equals(interactingNpc, npc))
+		{
+			interactingNpc = null;
+		}
+	}
+
+	private void resetHydras()
+	{
+		hydras.clear();
+		interactingNpc = null;
+	}
+
+	private void updateConfig()
```

Comment:
There's no reason for all of those fields - the overlay ones all add/remove an overlay, when config changed/on startup, you don't have to save that value after that. The bold one should probably set a field inside the `attackCounterOverlay`, rather than being gettered from here